### DISCUSSION TITLE
Is there any way to import win32api from pywin32 when running pythonnet in a c#/.NET runtime?

### DISCUSSION THREAD

**Discussion by tebjan** [pythonnet/pythonnet#2356] [Q&A] [answered]
I cannot get the win32api package to work there is always the "No module named 'win32api'" error. I've double-checked all paths, created a venv, and tested a script directly without running it from .NET, which works fine.

Is there anything I an try to get it to import? Place DLLs somewhere? Add special paths? Any ideas are welcome...

What I've tried so far was adding this path, where the two DLLs are located to the PYTHONPATH and PATH:

`venv\Lib\site-packages\pywin32_system32`

But it didn't help, unfortunately. What else needs to be in there to get pywin32 picked up by pythonnet? The `venv\Lib\site-packages\` and other paths like `Python\Python310\DLLs` are already added.

**Comment by tebjan** [comment]
@filmor said in the issue #2357:

> You have to make sure that the PATH and PYTHONPATH settings are correct.

But which paths need to be set to make pywin32 work correctly? the pywin32 module seems to be quite special. so far, other modules work fine in the venv.

**Reply by tebjan** [reply]
@filmor do you maybe have any additional ideas? I am not able to get this module to load in pythonnet. Would a minimal test project help to find the issue? Thanks!

**Comment by wxitcode** [comment, marked as answer]
[stackoverflow.com](https://stackoverflow.com/a/78292805/8624456)

1. Copy the two dlls from `venv\Lib\site-packages\pywin32_system32` to `C:\Windows\System32`

or **Run as administrator**

```
$ python .\venv\Scripts\pywin32_postinstall.py -install
```

2. Add 

 - `venv\Lib\site-packages\win32`
 - `venv\Lib\site-packages\win32\lib`
 - `venv\Lib\site-packages\Pythonwin`

to `PYTHONPATH`

**Reply by tebjan** [reply]
Super helpful, it works! With this, I got StreamDiffusion with TensorRT acceleration running at 45fps on my laptop with RTX 4090 GPU and 16GB VRAM: https://www.instagram.com/p/C5lbK5wMDQ2/
